src/skadi/sample.cc

```diff
@@ -448,14 +452,16 @@ sample::~sample() {
   delete cache_;
 }
 
-template <class coord_t> double sample::get(const coord_t& coord) {
+template <class coord_t> double sample::get(const coord_t& coord, tile_data& tile) {
   // check the cache and load
   auto lon = std::floor(coord.first);
   auto lat = std::floor(coord.second);
   auto index = static_cast<uint16_t>(lat + 90) * 360 + static_cast<uint16_t>(lon + 180);
 
-  // get the proper source of the data
-  const auto& tile = cache_->source(index);
+  if (index != tile.get_index()) {
```

Comment:
this is a bit tricky. you send an unitialized `tile_data` object to the function and it detects that its not initialized by looking at the index and seeing its wrong? shouldnt we check something else that is more obvious? maybe we could add a method to `bool tile_data::initilized() const { return index != TILE_COUNT; }` or what would be more obvious to me would be: `bool tile_data::initilized() const { return data != nullptr; }`

i just find this `if` with no comment very tricky because you have to figure out why the heck its checking the index

Reply:
ok so now i see why you are doing the trick with the index after i looked at the `get_all` change. caching. then i think the best thing to do here would be to add a comment above the `if` that says something like:

```suggestion
  // if the caller passed in a tile they were already working with we only fetch it if this coordinate needs a different tile
  if (index != tile.get_index()) {
```